lib/api.js

```diff
@@ -903,7 +903,7 @@ API.prototype.sendTxProposal = function(opts, cb) {
     feePerKb: opts.feePerKb,
     payProUrl: opts.payProUrl,
     type: opts.type,
-    outputs: opts.outputs,
+    outputs: _.cloneDeep(opts.outputs),
```

Comment:
because at line 910, the message is overwritten with the encrypted message. without cloning, the sendTxProposal() function will modify the client's copy of opts.